File: Datalake/datalakeclient/src/@types/enums/AccessType.ts
export enum AccessType {
	FIRST = -1,
	NOT = 0,
	USER = 1,
	ADMIN = 2,
}

export function AccessTypeDescription(type: AccessType) {
	switch (type) {
		case AccessType.FIRST: return 'Впервые'
		case AccessType.NOT: return 'Нет доступа'
		case AccessType.USER: return 'Пользователь'
		case AccessType.ADMIN: return 'Администратор'
		default: return '?'
	}
}

File: Contracts/Contracts.Public/Enums/AccessType.cs
namespace Datalake.Contracts.Public.Enums;

/// <summary>
/// Уровень доступа
/// </summary>
public enum AccessType : byte
{
	/// <summary>
	/// Правило не применяется
	/// </summary>
	None = 0,

	/// <summary>
	/// Разрешен только просмотр
	/// </summary>
	Viewer = 1,

	/// <summary>
	/// Разрешен только просмотр и запись значений
	/// </summary>
	Editor = 2,

	/// <summary>
	/// Доступ ко всем операциям
	/// </summary>
	Manager = 3,

	/// <summary>
	/// Доступа нет
	/// </summary>
	Denied = 4,

	/// <summary>
	/// Полный доступ
	/// </summary>
	Admin = 5,
}
